azurerm/resource_arm_securitycenter_contact.go

```diff
@@ -0,0 +1,153 @@
+// Invalid security contact name 'default0' was provided. Expected 'default1'
+// Message="Invalid security contact name 'default2' was provided. Expected 'default1'"
+
+func resourceArmSecurityCenterContact() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSecurityCenterContactCreateUpdate,
+		Read:   resourceArmSecurityCenterContactRead,
+		Update: resourceArmSecurityCenterContactCreateUpdate,
+		Delete: resourceArmSecurityCenterContactDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"email": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				//todo validation
+			},
+
+			"phone": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validation.NoZeroValues,
+			},
+
+			"alert_notifications": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+
+			"alerts_to_admins": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceArmSecurityCenterContactCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).securityCenterContactsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	contact := security.Contact{
+		ContactProperties: &security.ContactProperties{
+			Email: utils.String(d.Get("email").(string)),
+			Phone: utils.String(d.Get("phone").(string)),
+		},
+	}
+
+	if alertNotifications := d.Get("alert_notifications").(bool); alertNotifications {
+		contact.AlertNotifications = security.On
+	} else {
+		contact.AlertNotifications = security.Off
+	}
+
+	if alertNotifications := d.Get("alerts_to_admins").(bool); alertNotifications {
+		contact.AlertsToAdmins = security.AlertsToAdminsOn
+	} else {
+		contact.AlertsToAdmins = security.AlertsToAdminsOff
+	}
+
+	if d.IsNewResource() {
+		_, err := client.Create(ctx, "default1", contact)
+		if err != nil {
+			return fmt.Errorf("Error creating Security Center Contact: %+v", err)
+		}
+
+		resp, err := client.Get(ctx, "default1")
+		if err != nil {
+			return fmt.Errorf("Error reading Security Center Contact: %+v", err)
+		}
+		if resp.ID == nil {
+			return fmt.Errorf("Security Center Contact ID is nil")
+		}
+
+		d.SetId(*resp.ID)
+	} else {
+		_, err := client.Update(ctx, "default1", contact)
+		if err != nil {
+			return fmt.Errorf("Error updating Security Center Contact: %+v", err)
+		}
+	}
+
+	return resourceArmSecurityCenterContactRead(d, meta)
+}
+
+func resourceArmSecurityCenterContactRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).securityCenterContactsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	//id is in format of `/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/securityContacts/john`
+	//parseAzureResourceID doesn't support id without a resource group
```

Comment:
we should probably think about creating an overload which doesn't require it - there's a few instances of this now (key vault)